### PR TITLE
More description words in README.md for you to consider.

### PR DESCRIPTION
**Pull request by FeiYao-Edinburgh**
More description words in README.md for you to consider.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -36,13 +36,15 @@ data = [[ 0.80,  0.55,  0.22,  0.03],
 
 ```python
 from minisom import MiniSom    
-som = MiniSom(6, 6, 4, sigma=0.3, learning_rate=0.5) # initialization of 6x6 SOM
+# initialization of 6x6 SOM with an input of dimension of 4 for training.
+# The initial sigma and learning rate are 0.3 and 0.5, respectively. They decline by dividing (1+t/T) where T is half num_iteration.
```
Comment (JustGlowing) [inline Readme.md]: No need to be specific here, sigma and learning rate are the name of the parameters.

Comment (FeiYao-Edinburgh) [reply Readme.md]: Sure. I keep these for my personal use only. :)

File: Readme.md [hunk 2]
```diff
@@ -36,13 +36,15 @@ data = [[ 0.80,  0.55,  0.22,  0.03],
...
+MiniSom implements two types of training. The random training (implemented by the method `train_random`), where the model is trained picking random samples (i.e. rows) from your data, and the batch training (implemented by the method `train_batch`), where the samples are picked in the order they are stored. Both `train_random` and `train_batch` pick the iteration number of samples (i.e. rows) from your data for training. Ideally, they will approach similar results when iteration number is VERY large. Practically, I recommend using `train_batch` with the iteration number as multiple times of `len(data)` so as to ensure all samples within your data contribute equally.
 
-The weights of the network are randmly initialized by default. Two additional methods are provided to initialize the weights in a data driven fashion: `random_weights_init` and `pca_weights_init`.
+The weights of the network are randmly initialized by default. Two additional methods are provided to initialize the weights in a data driven fashion: `random_weights_init` and `pca_weights_init`. The `random_weights_init` is to randomly pick samples from your data instead of creating random numbers.
```
Comment (JustGlowing) [inline Readme.md]: The grammar here is not good enough. There are also some typos.

File: Readme.md [hunk 3]
```diff
@@ -36,13 +36,15 @@ data = [[ 0.80,  0.55,  0.22,  0.03],
...
+som = MiniSom(6, 6, 4, sigma=0.3, learning_rate=0.5)
 som.train_random(data, 100) # trains the SOM with 100 iterations
 ```
 
-MiniSom implements two types of training. The random training (implemented by the method `train_random`), where the model is trained picking random samples from your data, and the batch training (implemented by the method `train_batch`), where the samples are picked in the order they are stored.
+MiniSom implements two types of training. The random training (implemented by the method `train_random`), where the model is trained picking random samples (i.e. rows) from your data, and the batch training (implemented by the method `train_batch`), where the samples are picked in the order they are stored. Both `train_random` and `train_batch` pick the iteration number of samples (i.e. rows) from your data for training. Ideally, they will approach similar results when iteration number is VERY large. Practically, I recommend using `train_batch` with the iteration number as multiple times of `len(data)` so as to ensure all samples within your data contribute equally.
```
Comment (JustGlowing) [inline Readme.md]: I'd rather keep this short as all the details are specified in the documentation of the methods.

Also, I don't see the point in specifying that `train_batch` and `train_random` will yield similar results with enough iterations.

Comment (FeiYao-Edinburgh) [reply Readme.md]: > I'd rather keep this short as all the details are specified in the documentation of the methods.

It is surely fine to do this. I just wonder if those new to Python might prefer README.md first.



> Also, I don't see the point in specifying that `train_batch` and `train_random` will yield similar results with enough iterations.

Agreed!

File: Readme.md (hunk 2 above)
Comment (FeiYao-Edinburgh) [reply Readme.md]: Would be great if you improve the expression.

Comment (JustGlowing) [reply Readme.md]: I don't think it's worth the effort. Please don't focus on this documentation as I'm planning to reduce it to the bare minimum.

Comment (FeiYao-Edinburgh) [reply Readme.md]: Sorry to bother but really appreciate your reply!

Comment (JustGlowing) [reply Readme.md]: No problems.
